backend/internal/pkg/routes/car_test.go

```diff
@@ -76,10 +75,9 @@ func TestCreateCar(t *testing.T) {
 
 		resp := api.PostCtx(ctx, "/cars", testInput)
 		assert.Equal(t, http.StatusCreated, resp.Result().StatusCode)
-		respBody, _ := io.ReadAll(resp.Result().Body)
 
 		var car models.Car
-		err := json.Unmarshal(respBody, &car)
+		err := json.NewDecoder(resp.Result().Body).Decode(&car)
```

Comment:
Why do you use NewDecoder as opposed to Unmarshal? I am thinking since we are mocking in the test, the JSON is in data memory so Unmarshal would be better?